tests/behaviour/graql/graql_steps.py

```diff
@@ -16,108 +16,370 @@
-    try:
-        next(context.tx().query().delete(context.text))
-        assert False
-    except GraknClientException:
-        pass
+    assert_that(calling(next).with_args(context.tx().query().delete(query=context.text)), raises(GraknClientException))
 
 
 @step("graql delete; throws exception containing \"{exception}\"")
 def step_impl(context: Context, exception: str):
-    try:
-        next(context.tx().query().delete(context.text))
+    assert_that(calling(next).with_args(context.tx().query().insert(query=context.text)), raises(GraknClientException, exception))
+
+
+@step("get answers of graql insert")
+def step_impl(context: Context):
+    context.clear_answers()
+    context.answers = [answer for answer in context.tx().query().insert(query=context.text)]
+
+
+@step("get answers of graql match")
+def step_impl(context: Context):
+    context.clear_answers()
+    context.answers = [answer for answer in context.tx().query().match(query=context.text)]
+
+
+@step("graql match; throws exception")
+def step_impl(context: Context):
+    assert_that(calling(next).with_args(context.tx().query().match(query=context.text)), raises(GraknClientException))
+
+
+@step("get answer of graql match aggregate")
+def step_impl(context: Context):
+    context.clear_answers()
+    context.numeric_answer = next(context.tx().query().match_aggregate(query=context.text))
+
+
+@step("get answers of graql match group")
+def step_impl(context: Context):
+    context.clear_answers()
+    context.answer_groups = [group for group in context.tx().query().match_group(query=context.text)]
+
+
+@step("get answers of graql match group aggregate")
+def step_impl(context: Context):
+    context.clear_answers()
+    context.numeric_answer_groups = [group for group in context.tx().query().match_group_aggregate(query=context.text)]
+
+
+@step("answer size is: {expected_size:Int}")
+def step_impl(context: Context, expected_size: int):
+    assert_that(context.answers, has_length(expected_size), "Expected [%d] answers, but got [%d]" % (expected_size, len(context.answers)))
+
+
+class ConceptMatcher(object):
+
+    def matches(self, context: Context, concept: ConceptSubtype):
+        return False
+
+
+class TypeLabelMatcher(ConceptMatcher):
+
+    def __init__(self, label: str):
+        self.label = label
+
+    def matches(self, context: Context, concept: ConceptSubtype):
+        if concept.is_role_type():
+            return self.label == concept.get_scoped_label()
+        elif concept.is_type():
+            return self.label == concept.get_label()
+        else:
+            raise TypeError("A Concept was matched by label, but it is not a Type.")
+
+
+class AttributeMatcher(ConceptMatcher):
+
+    def __init__(self, type_and_value: str):
+        s = type_and_value.split(":")
+        assert_that(s, has_length(2), "[%s] is not a valid attribute identifier. It should have format \"type_label:value\"." % type_and_value)
+        self.type_label = s[0]
```

Comment:
@alexjpwalker this and next line could be shortened to:

```suggestion
        self.type_label, self.value = s
```